src/azure-cli/azure/cli/command_modules/sql/commands.py

```diff
@@ -220,6 +222,17 @@ def load_command_table(self, _):
 
         g.command('list-deleted', 'list_by_server')
 
+    restorable_dropped_managed_databases_operations = CliCommandType(
+        operations_tmpl='azure.mgmt.sql.operations#RestorableDroppedManagedDatabasesOperations.{}',
+        client_factory=get_sql_restorable_dropped_managed_databases_operations)
+
+    with self.command_group('sql midb',
+                            restorable_dropped_managed_databases_operations,
+                            client_factory=get_sql_restorable_dropped_managed_databases_operations) as g:
+
+        g.command('list-deleted', 'list_by_instance')
+        g.custom_command('show-deleted', 'show_deleted_database_mi')
```

Comment:
The two commands are a little werid. What about add a new parameter `--include-deleted` in existing show and list commands like storage blob command?

Reply:
That could work, however I wanted to keep cmdlets for MI similar to the SQLDB ones. As you can see couple of lines above SQLDB is using list-deleted. 

Other reason is that live and dropped databases are not same resource type. In PowerShell that breaks piping but I am not sure if this is an issue in CLI.

Reply:
I see. The reason I recommend such design is that we have similar implementation in storage commands and what we do is add an new parameters as what I have mentioned. And as far as I know, Powershell also has similar design. 
But as you said, in your scenario, it will break piping in Powershell. So there maybe some difference.
In this way, I am fine with the command if you already have similar design.